packages/devtools_shared/lib/src/memory/memory_json.dart

```diff
@@ -45,38 +60,41 @@ class MemoryJson {
 
   bool _memoryPayload;
 
-  /// JSON payload field "dartDevToolsScreen" has a value of "memory" e.g.,
+  /// JSON payload field "dart<T>DevToolsScreen" has a value of "memory" e.g.,
   ///   "dartDevToolsScreen": "memory"
   bool get isMemoryPayload => _memoryPayload;
 
   /// If data is empty check isMatchedVersion and isMemoryPayload to ensure the
   /// JSON file loaded is a memory file.
-  final List<HeapSample> data = [];
-
-  // TODO(terry): Expose encode/decode to Timeline too.
+  final List<T> data = [];
```

Comment:
not a new issue but I think we are overall trying to use the style of
final data = <T>[];

Reply:
Done.